frontend/app/components/date-picker-field.tsx

```diff
@@ -0,0 +1,428 @@
+              </>
+            ) : (
+              <>
+                {datePickerErrorMessages.month && <p>{datePickerErrorMessages.month}</p>}
+                {datePickerErrorMessages.day && <p>{datePickerErrorMessages.day}</p>}
+                {datePickerErrorMessages.year && <p>{datePickerErrorMessages.year}</p>}
+              </>
+            )}
+          </div>
+        )}
+        {helpMessagePrimary && (
+          <InputHelp
+            id={inputHelpMessagePrimaryId}
+            className={cn('mb-2', helpMessagePrimaryClassName)}
+            data-testid="date-picker-help-primary"
+          >
+            {helpMessagePrimary}
+          </InputHelp>
+        )}
+        <div className="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2">
+          {currentLanguage === 'fr' ? (
+            <>
+              {datePickerDay}
+              {datePickerMonth}
+              {datePickerYear}
+            </>
+          ) : (
+            <>
+              {datePickerMonth}
+              {datePickerDay}
+              {datePickerYear}
+            </>
+          )}
+        </div>
+        {helpMessageSecondary && (
+          <InputHelp
+            id={inputHelpMessageSecondaryId}
+            className={cn('mt-2', helpMessageSecondaryClassName)}
+            data-testid="date-picker-help-secondary"
+          >
+            {helpMessageSecondary}
+          </InputHelp>
+        )}
+      </fieldset>
+    </div>
+  );
+};
+
+interface DatePickerMonthProps {
+  ariaDescribedBy: string;
+  ariaErrorMessage?: string;
+  className?: string;
+  defaultValue: string;
+  disabled?: boolean;
+  id: string;
+  label: string;
+  name: string;
+  placeholder: string;
+  required?: boolean;
+}
+
+function DatePickerMonth({
+  ariaDescribedBy,
+  ariaErrorMessage,
+  className,
+  defaultValue,
+  disabled,
+  id,
+  label,
+  name,
+  placeholder,
+  required,
+}: DatePickerMonthProps) {
+  const { currentLanguage } = useLanguage();
+  const months = useMonths(currentLanguage ?? 'en');
+
+  const selectId = `date-picker-${id}-month`;
+  const wrapperId = `date-picker-${id}-month-wrapper`;
+  const inputLabelId = `date-picker-${id}-month-label`;
+  const inputOptionUnselectedId = `date-picker-${id}-month-option-unselected`;
+
+  return (
+    <div id={wrapperId} data-testid="date-picker-month">
+      <InputLabel id={inputLabelId} htmlFor={selectId} className="mb-2">
+        {label}
+      </InputLabel>
+      <select
+        aria-describedby={ariaDescribedBy}
+        aria-errormessage={ariaErrorMessage}
+        aria-invalid={!!ariaErrorMessage}
+        aria-labelledby={inputLabelId}
+        aria-required={required}
+        className={cn(inputBaseClassName, inputDisabledClassName, ariaErrorMessage && inputErrorClassName, className)}
+        data-testid="date-picker-month-select"
+        defaultValue={defaultValue}
+        disabled={disabled}
+        id={selectId}
+        name={name}
+        required={required}
```

Comment:
This could be simplified by moving some of these attributes into a `{...props}` spread.